akka-http-core/src/main/scala/akka/http/scaladsl/settings/ConnectionPoolSettings.scala

```diff
@@ -21,6 +24,26 @@ object PoolImplementation {
   case object New extends PoolImplementation
 }
 
+@ApiMayChange
+trait HostOverride {
```

Comment:
I think by now we have resolved the issue about confusion when loading configuration from a file first and then adopting it with code because we defined and implemented that hostOverrides will get all the updates made through code (because we now do `copy(hostOverrides = hostOverrides.map(...)` everywhere).

I wonder if the additional complexity of this extra indirection here (and also the `withOverrides` API) is still necessary at all, or we can now just live with the full API being available to users both from config files and code at all times?

Reply:
Agreed, updated